### ISSUE TITLE
kubectl reports different apiVersion then what I deployed

### ISSUE THREAD

**Issue by RudyBone** [kubernetes/kubectl#944]
My apologies if this issue/feature has been addressed elsewhere:

Running kubectl 1.14.5 against a 1.14.5 cluster.

I push a deployment yaml file up to the cluster with apiVersion: apps/v1

I do a kubectl get deployments <mydeployment> -o yaml and the apiVersion comes back as extensions/v1beta1

Using the --v=8 flag I can see that kubectl is actually issuing the REST call to the extensions/v1beta1 API.

kubectl api-resources lists deployments under both versions.

deployments are listed in both the .kube/cache/discovery/<server>/apps/v1 and the .kube/cache/discovery/<server>/extensions/v1beta1 serverresources.json file.

What is going on?

**Comment by eddiezane** [comment]
This was a compatibility translation due to the deprecation path for older APIs.

You can see that your deployment is actually part of `apps/v1` by hitting the endpoint manually.

```
kubectl get --raw /apis/apps/v1/deployments
```

A [thread on the mailing list](https://groups.google.com/g/kubernetes-dev/c/je0rjyfTVyc) says:

>In case you missed it in the 1.14.0 release notes, there are now concrete timelines for deprecated resources in the extensions/v1beta1, apps/v1beta1, and apps/v1beta2 API groups to no longer be served by default:
>
>DaemonSet, Deployment, and ReplicaSet resources will no longer be served from extensions/v1beta1, apps/v1beta1, or apps/v1beta2 by default in v1.16. Migrate to the apps/v1 API, available since v1.9. Existing persisted data can be retrieved/updated via the apps/v1 API.

/remove-kind bug
/kind support
/close

**Comment by eddiezane** [comment]
You should be able to specify the version you want directly as well with:

```
kubectl get deployments.v1.apps
```
